### PR TITLE
Charlesmchen/message view tweaks

### PR DESCRIPTION
**Pull request by charlesmchen**
* Unbound message bubble cache size and DRY up cache logic. 
* Remove obsolete "scrollLaterTimer" hack in messages view.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/Models/OWSMessagesBubblesSizeCalculator.m [hunk 1]
```diff
@@ -75,28 +79,40 @@ - (CGSize)messageBubbleSizeForMessageData:(id<JSQMessageData>)messageData
                               atIndexPath:(NSIndexPath *)indexPath
                                withLayout:(JSQMessagesCollectionViewFlowLayout *)layout
 {
+    id cacheKey = [self cacheKeyForMessageData:messageData];
```
Comment (charlesmchen) [inline Signal/src/Models/OWSMessagesBubblesSizeCalculator.m]: DRY.  The base class will do some redundant work updating the same cache for incoming and outgoing messages, but this is harmless and cheap.

File: Signal/src/Models/OWSMessagesBubblesSizeCalculator.m [hunk 2]
```diff
@@ -56,6 +56,10 @@ - (instancetype)init
         _referenceSystemMessageCell = [OWSSystemMessageCell new];
         _referenceUnreadIndicatorCell = [OWSUnreadIndicatorCell new];
         _referenceContactOffersCell = [OWSContactOffersCell new];
+
+        // Calculating message size is relatively expensive, so unbound the size of the cache.
+        self.cache.countLimit = 0;
+        self.cache.totalCostLimit = 0;
```
Comment (charlesmchen) [inline Signal/src/Models/OWSMessagesBubblesSizeCalculator.m]: I don't foresee this "hash-to-NSValues containing CGSizes" cache becoming that large in memory and its a huge perf hotspot.

File: Signal/src/ViewControllers/ConversationView/MessagesViewController.m [hunk 3]
```diff
@@ -228,7 +228,6 @@ @interface MessagesViewController () <AVAudioPlayerDelegate,
 @property (nonatomic) NSCache *messageAdapterCache;
 @property (nonatomic) BOOL userHasScrolled;
 @property (nonatomic) NSDate *lastMessageSentDate;
-@property (nonatomic) NSTimer *scrollLaterTimer;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/MessagesViewController.m]: This was an earlier hack to work around https://github.com/WhisperSystems/JSQMessagesViewController/pull/17.  Now that we've addressed the root cause, this timer can go away!

File: Signal/src/Models/OWSMessagesBubblesSizeCalculator.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/Models/OWSMessagesBubblesSizeCalculator.m]: Nice.

**Pull request merged**